test/unit/analysis/allocate_net_gen_test.py

```diff
@@ -271,7 +271,8 @@ def test_allocated_sums_match(example_1_pudl_tabl):
     # )
 
 
-def test_missing_energy_source():
+@pytest.fixture
+def example_2_pudl_tabl():
```

Comment:
Pulling this out into a fixture makes a lot of sense!

I think the new tests make the `example_1` tests basically redundant - `example_2` touches on the allocation sums, and they also touch on the fuel ratios too. The only thing is that `example_2` already has the bonus ESCs - so it tests a slightly less happy path. One thing we could do is have the following structure:

* nix `example_1`
* rename `example_2` as `one_plant_happy_path` or something, and *don't* put in the extra ESCs or PM codes just yet
* make new fixtures that add the extra ESCs or PM codes (`extra_esc_in_gf`, `extra_pm_in_bf`)
* use the fixtures in parameterized test that tests that the sums all make sense
* use the fixtures in parameterized test that tests that the ratios are correct for the happy path & the `extra_esc_in_gf` case
* add a separate SPECIAL TEST for the weird PM code behavior we found

I'll put a bit of time into trying that structure out to see if that works!